src/Symfony/Component/Runtime/RuntimeInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface RuntimeInterface
+{
+    /**
+     * Returns a closure that builds an object holding the userland app and returns it.
+     *
+     * @return \Closure(): object
+     */
+    public function getApp(\Closure $app): \Closure;
+
+    /**
+     * Returns a closure that runs the app and returns its exit status.
+     *
+     * @return \Closure(): int
+     */
+    public function getMain(object $app): \Closure;
```

Comment:
It sure is difficult. Here is a suggestion. Pick whatever you like from it. =)

```suggestion
     * Take an input closure and return an "app closure" that will build an object holding the 
     * userland app and return it. 
     *
     * The app closure MUST be able to be called with no arguments.
     *
     * For a Symfony application, the app closure will return a Kernel. 
     *
     * @return \Closure(): object
     */
    public function getApp(\Closure $input): \Closure;

    /**
     * Returns a closure that runs the application and returns its exit status.
     *
     * The closure MUST be able to be called with no arguments.
     *
     * For a Symfony HTTP application, the argument will be the Kernel and the 
     * closure will create a response and send it. 
     *
     * @return \Closure(): int
     */
    public function getMain(object $app): \Closure;
```

Reply:
Since this works not only with Symfony apps but with any really, I'd prefer not mentionning Symfony here...